homeassistant/components/goalzero/__init__.py

```diff
@@ -82,6 +90,24 @@ def __init__(self, api, coordinator, name, server_unique_id):
         self._server_unique_id = server_unique_id
         self._device_class = None
 
+    @property
+    def extra_state_attributes(self):
+        """Return the state attributes."""
+        attributes = {
+            ATTR_ATTRIBUTION: ATTRIBUTION,
+            "wifi_ssid": self.api.data.get("ssid"),
+            "ip_address": self.api.data.get("ipAddr"),
+        }
+        if (
+            ATTR_FOREIGN_ACCESSORY in self.api.data
+            and self.api.data[ATTR_FOREIGN_ACCESSORY] is not None
+        ):
+            attributes["accessory"] = self.api.data[ATTR_FOREIGN_ACCESSORY][ATTR_MODEL]
```

Comment:
What kind of value is this? If it's the model type it should be part of the device info and not state attributes. The state attributes should contain dynamic info related to the state of the entity. We have a limited number of exceptions like attribution.

Reply:
In my case, there is an accessory that enables better charging which can be removed at will. How should it be done if we already have info for model and firmware for the main device?

Reply:
We could add a separate device for the accessory.

https://developers.home-assistant.io/docs/device_registry_index#what-is-a-device

Does the accessory have any useful info that we want to expose as sensors?

Reply:
No. The yeti battery just reports that an accessory is connected and its firmware. That's it.

Reply:
What's the reason we want to expose the accessory model and firmware to the user if that's all the information there is?

Reply:
No other reason except for complete disclosure for the user. I take it that you want this part to be left out leaving the wifi network and ip address?

Reply:
Wifi ssid and ip address should be separate sensors disabled by default. We shouldn't add any new state attributes except attribution here.

Reply:
Ok. Would it be acceptable to append the presence of an accessory to the `sw_version`?

Reply:
It's not really part of that info, so I'd say no. If we want to expose info from the accessory it should be a separate device, since that's what it is, as far as I can tell.